### ISSUE TITLE
Change from src to another path?

### ISSUE THREAD

**Issue by mackelito** [Swiip/generator-gulp-angular#243]
Hey!

Im working on a site but need to change so that files that end up in the "src" folder ends up in the root instead.

Have moved it manually but can't get gulp to work... ?
Or perhaps this i a bad idea??

**Comment by marani** [comment]
For generated project of current version ( <= 0.8.1), you probably need to do roughly the following:
- find-replace everything with `'src/'` in the gulp files in `/gulp` folder and change them to either `''` or `'./'`
- do the same thing in `index.html`.
- take care of relative paths when importing styles, also some places in `gulp/build.js`

For future versions, use `yo gulp-angular --app-path='.'`. See more in [README](https://github.com/Swiip/generator-gulp-angular#yo-options). The path configuration feature is on master branch, hasn't been published to npm yet. You can try to install the generator from master & use the above path option.

For generated project of the future versions, these path configurations are centralized in [gulpfile.js](https://github.com/Swiip/generator-gulp-angular/blob/master/app/templates/_gulpfile.js), we just need to change from there.

Bottom line, that's possible to do, but I think putting stuff in `src` folder to root is kind of messy.

**Comment by mackelito** [comment]
We are using spring boot as the server and it sets a context path (the www root) everything below that path gets public.

That means that we will not have access to bower_components etc.. When developing 😕

What is the main reason for having the /src/ structure?

**Comment by mackelito** [comment]
Btw.. how do I run yo gulp-angular with the --app-path='.' config?
Or is that not possible yet?

**Comment by marani** [comment]
Yup that's not possible yet, as mentioned above, it's on master but not yet on npm.

**Comment by mackelito** [comment]
Ah I see.. any idea when it will be possible to use?
I'm about to set things up so should I wait or try to do it manually? :)

**Comment by marani** [comment]
I don't really understand the `bower_components` problem part, is it because everything is in `src` ?

Having all our domain-specific code in `src` is just a way of structuring the app that has some benefits:
- First it makes thing easier to find (most of the time will be spent on writing code for that folder). If we merge this with root then developers might have to search from more things (dot files, .tmp folder...) just to find useful source code.
- That also makes quick file switching (ctrl/cmd + P in sublime) faster if you only open that `src` folder.

**Comment by mackelito** [comment]
The problem is that that if we sets src as the root (when you visit eg. localhost) you will get the index.html but the bower_components that are located at the same level as the src will not be accessible.

**Comment by marani** [comment]
Ahh got it.
Doing it manually wouldn't take much time. If the project is urgent I recommend taking the above steps & do it now :)

I'm not really sure when new version will be released, probably very soon. @Swiip

**Comment by marani** [comment]
You can also download from master branch then `npm install` then `npm link` & try it.

**Comment by wgorder** [comment]
I am using Spring Boot without any issues.  The link below is a bit out of date and I cant share my production app, but you can get an idea. The trick is when the 'localDev' profile is active spring boot reads things directly off of the file system, when building, the gradle task jars the dist directory in the client project (the project generated with this generator) and puts in a static directory that way spring boot will automatically serve it (the client.jar is included as a dependency for server project).  I had to make a few additional changes that I have not made there to make things run perfectly but it will maybe give you a start.

https://github.com/wgorder/acme-ui

**Comment by mackelito** [comment]
We solved it in another way but thx for the reply! :)
